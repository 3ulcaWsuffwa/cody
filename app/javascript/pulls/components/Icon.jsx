// @flow

import React from "react";

function sizeModifier(size) {
  switch (size) {
    case "small":
    case "medium":
    case "large":
      return `is-${size}`;
    default:
      return "";
  }
}

const Icon = ({
  icon,
  size
}: {
  icon: string,
  size: ?("small" | "medium" | "large")
}) => (
  <span className={`icon ${sizeModifier(size)}`}>
    <i className={`fas fa-${icon}`} />
  </span>
);

export default Icon;
