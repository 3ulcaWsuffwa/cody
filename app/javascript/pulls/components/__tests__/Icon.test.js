import React from "react";
import { render } from "@testing-library/react";
import Icon from "../Icon";

test("Icon snapshot test", () => {
  const component = render(<Icon icon="user-circle" />);
  let fragment = component.asFragment();
  expect(fragment).toMatchSnapshot();
});
