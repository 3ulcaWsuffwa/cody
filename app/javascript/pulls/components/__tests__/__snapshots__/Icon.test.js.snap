// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon snapshot test 1`] = `
<DocumentFragment>
  <span
    class="icon "
  >
    <i
      class="fas fa-user-circle"
    />
  </span>
</DocumentFragment>
`;
